### DISCUSSION TITLE
How can I create a reference based on multiple foreign fields?

### DISCUSSION THREAD

**Discussion by tomerh2001** [typegoose/typegoose#941] [Q&A] [answered]
I have the following mongodb query in my code:
```typescript
await ActivityPlatformModel.find({
	activityId: context.activity._id,
	platformId: context.platform.id,
});
```

It is basically redundant since I can use `Ref` and spare myself the manual query - so I tried just that, I created the following class:
```typescript
@plugin(autoPopulate)
export class ActivityPlatformReference {
	@prop({
		ref: () => ActivityPlatform,
		localField: 'activityId',
		foreignField: '_id',
		autopopulate: true,
	})
		platform?: ActivityPlatform[];

	@prop({type: () => Types.ObjectId})
		activityId!: Types.ObjectId;

	@prop({type: () => String})
		platformId!: PlatformId;
}
```

But I can't seem to find a way to specify that I need both `activityId` and `platformId` as my `localField`s / how to specify the matching foreign fields.

How does one do multi-field references in Mongoose/Typegoose?

**Comment by hasezoey** [comment, marked as answer]
i have not personally needed to do this, but you might wanna look into the [`match`](https://typegoose.github.io/typegoose/docs/api/decorators/prop#virtual-populate-match) option and read [Populate Virtuals: The Match Option](https://mongoosejs.com/docs/populate.html#match)

**Reply by tomerh2001** [reply]
I'm sure I'm not the first that needs a multi key reference...

**Reply by hasezoey** [reply]
you are not the first in the [mongoose space](https://stackoverflow.com/questions/43551953/mongoose-populate-virtuals-with-multiple-local-foreign-key-pairs), but in typegoose you are the first who at least asked. and i personally have not needed to use that yet.
